src/CommunityToolkit/Xamarin.CommunityToolkit/Views/BadgeView/BadgeView.shared.cs

```diff
@@ -272,7 +272,7 @@ void UpdateBadgeViewPlacement(bool force = false)
 			{
 				const double Padding = 6;
 				var size = Math.Max(BadgeText.Height, BadgeText.Width) + Padding;
-				BadgeIndicatorBackground.HeightRequest = size;
+				BadgeIndicatorBackground.HeightRequest = BadgeText.Height + Padding;
```

Comment:
it doesn't seem to be a proper fix cuz indicator should be stuck to the corner. Now it's above

Reply:
Also, currently, it's supposed to have equal width and height. So for allowing non-square / non-circle shapes, I think, we should expose a new property

Reply:
I think current margins calculations are supposing to not have a big difference between Width and Height.

Reply:
``var size = Math.Max(BadgeText.Height, BadgeText.Width) + Padding```
It always takes "greatest" parameter now